compiler/qsc_rir/src/passes/ssa_check.rs

```diff
@@ -0,0 +1,230 @@
+                | Instruction::Icmp(_, Operand::Literal(_), Operand::Variable(var), _)
+                | Instruction::LogicalNot(Operand::Variable(var), _)
+                | Instruction::LogicalAnd(Operand::Variable(var), Operand::Literal(_), _)
+                | Instruction::LogicalAnd(Operand::Literal(_), Operand::Variable(var), _)
+                | Instruction::LogicalOr(Operand::Variable(var), Operand::Literal(_), _)
+                | Instruction::LogicalOr(Operand::Literal(_), Operand::Variable(var), _)
+                | Instruction::BitwiseNot(Operand::Variable(var), _)
+                | Instruction::BitwiseAnd(Operand::Variable(var), Operand::Literal(_), _)
+                | Instruction::BitwiseAnd(Operand::Literal(_), Operand::Variable(var), _)
+                | Instruction::BitwiseOr(Operand::Variable(var), Operand::Literal(_), _)
+                | Instruction::BitwiseOr(Operand::Literal(_), Operand::Variable(var), _)
+                | Instruction::BitwiseXor(Operand::Variable(var), Operand::Literal(_), _)
+                | Instruction::BitwiseXor(Operand::Literal(_), Operand::Variable(var), _) => {
+                    add_use(var.variable_id, block_id, idx);
+                }
+
+                // Double variable
+                Instruction::Add(Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::Sub(Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::Mul(Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::Sdiv(Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::Srem(Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::Shl(Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::Ashr(Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::Icmp(_, Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::LogicalAnd(Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::LogicalOr(Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::BitwiseAnd(Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::BitwiseOr(Operand::Variable(var1), Operand::Variable(var2), _)
+                | Instruction::BitwiseXor(Operand::Variable(var1), Operand::Variable(var2), _) => {
+                    add_use(var1.variable_id, block_id, idx);
+                    add_use(var2.variable_id, block_id, idx);
+                }
+
+                // Multiple variables
+                Instruction::Call(_, vals, _) => {
+                    for val in vals {
+                        if let Operand::Variable(var) = val {
+                            add_use(var.variable_id, block_id, idx);
+                        }
+                    }
+                }
+                Instruction::Phi(vals, _) => {
```

Comment:
In other parts of the code base we refer to the `Vec<_>` component of phi nodes as args. We should be consistent one way or another.